magento-research/pwa-studio
# [Documentation] Peregrine hooks reference

## Description

* Add JSDoc annotations to the Peregrine hooks.
* Add ability to create markdown content from JSDoc annotations
* Create reference docs from JSDoc annotations

## Related Issue

Closes #1237 

## Verification Steps
0. Make sure you have `ruby` and `bundler` installed
1. Navigate to the `pwa-devdocs` project directory
2. Run `npm run build-reference-doc-snippets`
3. Verify that the following files exist in `src/_includes/_auto-generated/peregrine/src/hooks`:
    * `useApolloContext.md`
    * `useDropdown.md`
    * `useEventListener.md`
    * `useQuery.md`
    * `useQueryResult.md`
    * `useSearchParam.md`
    * `useWindowSize.md`
4. Run `npm run develop`
5. Verify that the site builds successfully.
6. Navigate to the Peregrine section (`/peregrine/`)
7. Verify the existence of the **Custom React Hooks** section in the left hand table of contents
8. Verify each of the generated hooks have corresponding pages in that section and contain reference data from the source code.

## How Have YOU Tested this?

See **Verification Steps**

## Checklist:
